lib/project-data.ts

```diff
@@ -30,7 +30,16 @@ export class ProjectData implements IProjectData {
 	public projectDir: string;
 	public platformsDir: string;
 	public projectFilePath: string;
-	public projectId: string;
+	public projectIdentifiers: IProjectIdentifier;
+	get projectId(): string {
+		this.warnProjectId();
+		return this.projectIdentifiers.ios;
+	}
+	set projectId(identifier: string) {
+		this.warnProjectId();
+		this.projectIdentifiers.ios = identifier;
+		this.projectIdentifiers.android = identifier;
```

Comment:
is this a valid android identifier?

Reply:
No the validation happens at a later stage.

Reply:
it's up to you, but as I mentioned earlier validation should happen as early as possible so other modules can be sure validation is already done